pkg/manifests/external_dns.go

```diff
@@ -347,6 +435,22 @@ func newExternalDNSDeployment(conf *config.Config, externalDnsConfig *ExternalDn
 
 	serviceAccount := externalDnsConfig.serviceAccountName
 
+	deploymentArgs := []string{
+		"--provider=" + externalDnsConfig.provider.string(),
+		"--interval=" + conf.DnsSyncInterval.String(),
+		"--txt-owner-id=" + conf.ClusterUid,
+		"--txt-wildcard-replacement=" + txtWildcardReplacement,
+	}
+
+	resourceTypeArgs := make([]string, 0, 3)
```

Comment:
why are we initializing with a capacity of 3? That's the max amount of deployment args? Not worth that optimization, easier to mess up. Just initialize to a standard empty slice

Reply:
it's the max number of args specifically for the types of resources we support - just small perf improvement, not sure what it messes up

Reply:
using empty slice now